### PR TITLE
Add overload for time series checkpoint API that takes a stream

### LINKED ISSUE #3804: Add overload for time series checkpoint API that takes a stream
This is needed is scenarios where the user is running the time series prediction on a blob where they don't have files but memory streams.

### PR DESCRIPTION
**Pull request by codemzs**
fixes #3804

### CODE REVIEW

File: src/Microsoft.ML.TimeSeries/PredictionFunction.cs [hunk 1]
```diff
@@ -76,15 +76,34 @@ public sealed class TimeSeriesPredictionFunction<TSrc, TDst> : PredictionEngineB
...
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// This is an example for checkpointing time series that detects change point using Singular Spectrum Analysis (SSA) model.
+        /// [!code-csharp[MF](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
+        public void CheckPoint(IHostEnvironment env, Stream stream)
+        {
```
Comment (eerhardt) [inline src/Microsoft.ML.TimeSeries/PredictionFunction.cs]: Are there any parameter checks that are needed?

File: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms.TimeSeries;
+
+namespace Samples.Dynamic
+{
+    public static class DetectChangePointBySsaStream
+    {
+        // This example creates a time series (list of Data with the i-th element corresponding to the i-th time slot). 
+        // It demostrates stateful prediction engine that updates the state of the model and allows for saving/reloading.
```
Comment (eerhardt) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs]: spelling - `demostrates` => `demonstrates`

File: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+            // Convert data to IDataView.
+            var dataView = ml.Data.LoadFromEnumerable(data);
+
+            // Setup SsaChangePointDetector arguments
+            var inputColumnName = nameof(TimeSeriesData.Value);
+            var outputColumnName = nameof(ChangePointPrediction.Prediction);
+
+            // Train the change point detector.
+            ITransformer model = ml.Transforms.DetectChangePointBySsa(outputColumnName, inputColumnName, 95, 8, TrainingSize, SeasonalitySize + 1).Fit(dataView);
```
Comment (eerhardt) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs]: Where do the numbers `95` and `8` come from?

File: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            // 300     1     256.83    0.01    22130423541.93    <-- alert is on, note that delay is expected
+            // 400     0     326.55    0.04    241162710263.29
+            // 500     0     364.82    0.08    597660527041.45   <-- saved to disk
+
+            // Now we demonstrate saving and loading the model.
+
+            // Save the model that exists within the prediction engine.
+            // The engine has been updating this model with every new data point.
+            var modelPath = "model.zip";
+            using (var stream = File.Create(modelPath))
```
Comment (eerhardt) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs]: Should we use a different kind of Stream than a file? That way this sample is different than the other one that also writes to a file.

Comment (codemzs) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs]: Good idea!

**Pull request merged**